common/enableRounds/enableRoundsHandler.go

```diff
@@ -0,0 +1,52 @@
+package enableRounds
+
+import (
+	"fmt"
+
+	"github.com/ElrondNetwork/elrond-go-core/core/atomic"
+	"github.com/ElrondNetwork/elrond-go/config"
+)
+
+const checkValueOnExecByCallerName = "CheckValueOnExecByCaller"
+
+type enableRoundsHandler struct {
+	*flagsHolder
+}
+
+// NewEnableRoundsHandler creates a new enable rounds handler instance
+func NewEnableRoundsHandler(args config.RoundConfig) (*enableRoundsHandler, error) {
+	checkValueOnExecByCaller, err := getRoundConfig(args, checkValueOnExecByCallerName)
+	if err != nil {
+		return nil, err
+	}
+
+	return &enableRoundsHandler{
+		flagsHolder: &flagsHolder{
+			checkValueOnExecByCaller: checkValueOnExecByCaller,
+		},
+	}, nil
+}
+
+func getRoundConfig(args config.RoundConfig, configName string) (*roundFlag, error) {
+	for _, activationRound := range args.RoundActivations {
+		if activationRound.Name == configName {
+			return &roundFlag{
+				round:   activationRound.Round,
+				flag:    &atomic.Flag{},
+				options: activationRound.Options,
+			}, nil
+		}
+	}
+
+	return nil, fmt.Errorf("%w for config %s", errMissingRoundActivation, configName)
+}
+
+// CheckRound should be called whenever a new round is known. It will trigger the updating of all containing round flags
+func (handler *enableRoundsHandler) CheckRound(round uint64) {
```

Comment:
I would prefer CheckRound since it is clear enough and resembles the EpochNotifier.CheckEpoch
This method will be called once when the node starts to process a received block or whenever it wants to propose a new one. The method will update **all current and future flags** (it should be slight optimal this approach as opposed the old handler and for loop implementation -> we are phasing that implementation out, also)